src/java/org/apache/cassandra/index/sai/utils/PrimaryKey.java

```diff
@@ -243,6 +287,8 @@ private void loadDeferred()
         }
     }
 
+    boolean isTokenOnly();
```

Comment:
nit: Perhaps this could just be a default method that returns false, then only the token-only case would have to override?